Build failure: fbida

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix-build -A fbida`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317990774

### Relevant log output

```console
/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/bash: line 1: /bin/echo: No such file or directory
gcc -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith -Wunused -DVERSION="2.14" -I. -Wno-pointer-sign -Ijpeg/62 -I/nix/store/1cnvyk57f3wqlmbpidz0x2824ip9f8ly-pixman-0.46.4/include/pixman-1 -Wp,-MD,mk/exiftran.tmp -c -o exiftran.o exiftran.c
gcc -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith -Wunused -DVERSION="2.14" -I. -Wno-pointer-sign -Ijpeg/62 -I/nix/store/1cnvyk57f3wqlmbpidz0x2824ip9f8ly-pixman-0.46.4/include/pixman-1 -Wp,-MD,mk/genthumbnail.tmp -c -o genthumbnail.o genthumbnail.c
genthumbnail.c: In function 'compress_thumbnail':
genthumbnail.c:128:19: error: too many arguments to function 'jpeg_std_error'; expected 0, have 1
  128 |     thc.dst.err = jpeg_std_error(&thc.err);
      |                   ^~~~~~~~~~~~~~ ~~~~~~~~
In file included from genthumbnail.c:8:
jpeg/62/jpeglib.h:884:33: note: declared here
  884 | EXTERN(struct jpeg_error_mgr *) jpeg_std_error
      |                                 ^~~~~~~~~~~~~~
jpeg/62/jpeglib.h:895:5: error: too many arguments to function 'jpeg_CreateCompress'; expected 0, have 3
  895 |     jpeg_CreateCompress((cinfo), JPEG_LIB_VERSION, \
      |     ^~~~~~~~~~~~~~~~~~~ ~~~~~~~
genthumbnail.c:129:5: note: in expansion of macro 'jpeg_create_compress'
  129 |     jpeg_create_compress(&thc.dst);
      |     ^~~~~~~~~~~~~~~~~~~~
jpeg/62/jpeglib.h:900:14: note: declared here
  900 | EXTERN(void) jpeg_CreateCompress JPP((j_compress_ptr cinfo,
      |              ^~~~~~~~~~~~~~~~~~~
genthumbnail.c:138:5: error: too many arguments to function 'jpeg_set_defaults'; expected 0, have 1
  138 |     jpeg_set_defaults(&thc.dst);
      |     ^~~~~~~~~~~~~~~~~ ~~~~~~~~
jpeg/62/jpeglib.h:914:14: note: declared here
  914 | EXTERN(void) jpeg_set_defaults JPP((j_compress_ptr cinfo));
      |              ^~~~~~~~~~~~~~~~~
genthumbnail.c:139:5: error: too many arguments to function 'jpeg_start_compress'; expected 0, have 2
  139 |     jpeg_start_compress(&thc.dst, TRUE);
      |     ^~~~~~~~~~~~~~~~~~~ ~~~~~~~~
jpeg/62/jpeglib.h:936:14: note: declared here
  936 | EXTERN(void) jpeg_start_compress JPP((j_compress_ptr cinfo,
      |              ^~~~~~~~~~~~~~~~~~~
genthumbnail.c:142:9: error: too many arguments to function 'jpeg_write_scanlines'; expected 0, have 3
  142 |         jpeg_write_scanlines(&thc.dst, (void*)ida_image_scanline(img, i), 1);
      |         ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~
jpeg/62/jpeglib.h:938:20: note: declared here
  938 | EXTERN(JDIMENSION) jpeg_write_scanlines JPP((j_compress_ptr cinfo,
      |                    ^~~~~~~~~~~~~~~~~~~~
genthumbnail.c:144:5: error: too many arguments to function 'jpeg_finish_compress'; expected 0, have 1
  144 |     jpeg_finish_compress(&(thc.dst));
      |     ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
jpeg/62/jpeglib.h:941:14: note: declared here
  941 | EXTERN(void) jpeg_finish_compress JPP((j_compress_ptr cinfo));
      |              ^~~~~~~~~~~~~~~~~~~~
genthumbnail.c:145:5: error: too many arguments to function 'jpeg_destroy_compress'; expected 0, have 1
  145 |     jpeg_destroy_compress(&(thc.dst));
      |     ^~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
jpeg/62/jpeglib.h:905:14: note: declared here
  905 | EXTERN(void) jpeg_destroy_compress JPP((j_compress_ptr cinfo));
      |              ^~~~~~~~~~~~~~~~~~~~~
make: *** [mk/Compile.mk:60: genthumbnail.o] Error 1
error: Cannot build '/nix/store/q784g9ngbi8g0ki5g7lg9r42b4wlxqgc-fbida-2.14.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/xgw16cy2qf871q5xnkia3p7hmd0syhbq-fbida-2.14
       Last 25 log lines:
       > genthumbnail.c:139:5: error: too many arguments to function 'jpeg_start_compress'; expected 0, have 2
       >   139 |     jpeg_start_compress(&thc.dst, TRUE);
       >       |     ^~~~~~~~~~~~~~~~~~~ ~~~~~~~~
       > jpeg/62/jpeglib.h:936:14: note: declared here
       >   936 | EXTERN(void) jpeg_start_compress JPP((j_compress_ptr cinfo,
       >       |              ^~~~~~~~~~~~~~~~~~~
       > genthumbnail.c:142:9: error: too many arguments to function 'jpeg_write_scanlines'; expected 0, have 3
       >   142 |         jpeg_write_scanlines(&thc.dst, (void*)ida_image_scanline(img, i), 1);
       >       |         ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~
       > jpeg/62/jpeglib.h:938:20: note: declared here
       >   938 | EXTERN(JDIMENSION) jpeg_write_scanlines JPP((j_compress_ptr cinfo,
       >       |                    ^~~~~~~~~~~~~~~~~~~~
       > genthumbnail.c:144:5: error: too many arguments to function 'jpeg_finish_compress'; expected 0, have 1
       >   144 |     jpeg_finish_compress(&(thc.dst));
       >       |     ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
       > jpeg/62/jpeglib.h:941:14: note: declared here
       >   941 | EXTERN(void) jpeg_finish_compress JPP((j_compress_ptr cinfo));
       >       |              ^~~~~~~~~~~~~~~~~~~~
       > genthumbnail.c:145:5: error: too many arguments to function 'jpeg_destroy_compress'; expected 0, have 1
       >   145 |     jpeg_destroy_compress(&(thc.dst));
       >       |     ^~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
       > jpeg/62/jpeglib.h:905:14: note: declared here
       >   905 | EXTERN(void) jpeg_destroy_compress JPP((j_compress_ptr cinfo));
       >       |              ^~~~~~~~~~~~~~~~~~~~~
       > make: *** [mk/Compile.mk:60: genthumbnail.o] Error 1
```

### Additional context

Possibly related https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/72ym7zz89wa91k3g4vz2g97pma1n51zy-xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`
```

### Notify maintainers

@pSub 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project. I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

--- a/pkgs/by-name/fb/fbida/function-parameters.patch
+++ b/pkgs/by-name/fb/fbida/function-parameters.patch
@@ -1,0 +1,16 @@
+diff --git a/jpeg/62/jpeglib.h b/jpeg/62/jpeglib.h
+index d1be8dd..38436ef 100644
+--- a/jpeg/62/jpeglib.h
++++ b/jpeg/62/jpeglib.h
+@@ -814,11 +814,7 @@ typedef JMETHOD(boolean, jpeg_marker_parser_method, (j_decompress_ptr cinfo));
+  * Note JPP requires double parentheses.
+  */
+ 
+-#ifdef HAVE_PROTOTYPES
+ #define JPP(arglist)	arglist
+-#else
+-#define JPP(arglist)	()
+-#endif
+ 
+ 
+ /* Short forms of external names for systems with brain-damaged linkers.

--- a/pkgs/by-name/fb/fbida/package.nix
+++ b/pkgs/by-name/fb/fbida/package.nix
@@ -41,6 +41,8 @@ stdenv.mkDerivation rec {
       url = "https://git.kraxel.org/cgit/fbida/patch/?id=1bb8a8aa29845378903f3c690e17c0867c820da2";
       sha256 = "0n5vqbp8wd87q60zfwdf22jirggzngypc02ha34gsj1rd6pvwahi";
     })
+    # Prevents using function declaration without explicit parameters.
+    ./function-parameters.patch
   ];
 
   nativeBuildInputs = [

